### PR TITLE
Support multiple profiles

### LINKED ISSUE #27760: Some Quarkus mode are enforcing a specific profile making it impossible to use another config profile
### Describe the bug

I am trying to use remote dev mode on openshift, I have a specific profile for openshift named `openshift`.

When I activate remote dev mode, `dev` profile gets activated breaking the configuration which needs the `openshift` profile to work.

### Expected behavior

Have both `dev` and `openshift` profiles enabled when it was used at build time or when it is set with `quarkus.profile`.

### Actual behavior

Only `dev` gets activated

### How to Reproduce?

I can prepare a reproducer if needed.

### LINKED ISSUE #28176: quarkus.profile is fixed at build time while it is documented as runtime
### Describe the bug

quarkus profile cannot be changed at runtime. It always stick to the value configured at build time.
I don't know if this is expected or not, 
but in the documentation, it is documented as a runtime value (without the lock). 

Also I think it should be a good thing to be able to set the profile at runtime (while I understand this would only affect runtime config values, and the profile used during the build would affect build time values)

Verified on 2.12.3.Final;

### LINKED ISSUE #29944: QUARKUS_TEST_PROFILE env variable is not honored
### Describe the bug

Issue:
Can't change test profile using environmente variable *QUARKUS_TEST_PROFILE*, as described at https://quarkus.io/guides/continuous-testing#quarkus-test-dev-testing-test-config_quarkus.test.profile

Reproduction:

Actual:
```
Green test: *mvn clean test -Dquarkus.test.profile=my_profile*
Red test: *QUARKUS_TEST_PROFILE=my_profile mvn clean test*
```

Expected:
```
Green test: *mvn clean test -Dquarkus.test.profile=my_profile*
Green test: *QUARKUS_TEST_PROFILE=my_profile mvn clean test*
```

### Expected behavior

See 'Describe the bug'

### Actual behavior

See 'Describe the bug'

### How to Reproduce?

https://github.com/siudeks/repro/tree/quarkus-test-profile

### Output of `uname -a` or `ver`

Linux PC1 5.15.79.1-microsoft-standard-WSL2 #1 SMP Wed Nov 23 01:01:46 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

java 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

maven

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by radcortez**
Initial work to support #27760.

- Fixes https://github.com/quarkusio/quarkus/issues/27760
- Fixes https://github.com/quarkusio/quarkus/issues/28176
- Fixes https://github.com/quarkusio/quarkus/issues/29944

Get rid of calls to `ProfileManager#getActiveProfile` and rely on the profiles list coming from SmallRye Config.

### CODE REVIEW

**Comment by geoand** [comment]
What's the status of this?

Seems important to have :)

**Comment by ia3andy** [comment]
I am waiting for the CI to pass to give it a try

**Comment by geoand** [comment]
CI has passed 😉

**Comment by radcortez** [comment]
Would like some feedback from @ia3andy, since this was mainly to fix #27760.

**Comment by ia3andy** [comment]
I've tried it with different cases and it seems to work fine (and as I expect it to work).

**Comment by radcortez** [comment]
Approve? :)

**Review by ia3andy — APPROVED** [review]
LGTM!

I am not sure why this is called "initial work..." instead of "support multiple profiles"

**Comment by ia3andy** [comment]
As this PR is side-effect friendly, I think it would be wise to have another review.

**Comment by geoand** [comment]
I'm going to rebase this on `main` to get another CI run.

**Comment by geoand** [comment]
I think we should get this in `2.14`. @gsmet WDYT?

**Comment by ia3andy** [comment]
@gsmet up

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/ProfileManager.java
Comment (gastaldi) [inline core/runtime/src/main/java/io/quarkus/runtime/configuration/ProfileManager.java:43]: Add a Javadoc explaining why this is deprecated and what's the alternative?

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/QuarkusConfigFactory.java
Comment (gastaldi) [inline core/runtime/src/main/java/io/quarkus/runtime/configuration/QuarkusConfigFactory.java:29]: Is it intentional to no longer add default/discovered converters and interceptors here?

**Review by gastaldi — COMMENTED** [review]
Added some comments

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/QuarkusConfigFactory.java
Comment (radcortez) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/QuarkusConfigFactory.java:29]: The call to `ConfigUtils.emptyConfigBuilder()` provides a builder with discovered converters enabled, so need to do it again.

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/ProfileManager.java
Comment (radcortez) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/ProfileManager.java:48]: Added javadoc.

**Review by gastaldi — APPROVED** [review]

**Comment by ia3andy** [comment]
@radcortez I think with two approvals we can mege?

**Comment by ia3andy** [comment]
This being a big change I think making it Quarkus 3.0 might be a good idea no?

**Comment by radcortez** [comment]
> @radcortez I think with two approvals we can merge?

Sure, just wanted to make sure that @gsmet was ok with this.

> This being a big change I think making it Quarkus 3.0 might be a good idea no?

Hum, I thought you needed this sooner. I mean, it is a considerable change, but not so big that it requires a major version.

**Comment by ia3andy** [comment]
@radcortez, I think @gsmet had plenty of time to give an opinion on this :) I am not sure if the CI is considered as passed or not, but if it is, you can merge

**Comment by radcortez** [comment]
The CI failure is unrelated. A timeout issue.

Well, I'll be merging this then.

**Pull request merged**
